climada/hazard/landslide.py

```diff
@@ -21,160 +21,99 @@
 __all__ = ['Landslide']
 
 import logging
-from pathlib import Path
-import glob
 
-import shlex
-import subprocess
+import geopandas as gpd
+import numpy as np
+import pandas as pd
 from scipy import sparse
 from scipy.stats import binom
-import geopandas
-import pyproj
-import matplotlib.pyplot as plt
-import rasterio
-from rasterio.windows import Window
-import requests
-import numpy as np
-from haversine import haversine
+import shapely
 
-from climada import CONFIG
 from climada.hazard.base import Hazard
-from climada.util.constants import SYSTEM_DIR as LS_FILE_DIR
 
 LOGGER = logging.getLogger(__name__)
 
 HAZ_TYPE = 'LS'
 
 
-"""for future: implement a function that downloads COOLR data by command, not manually"""
-# def get_coolr_shp(save_path=os.getcwd()):
-#    """for LS_MODEL[0]: download most up-to-date version of historic LS records from
-#    global landslide catalog (COOLR of NASA) in shape-file format (zip)"""
-
-#   trials didn't work.
-#    url = 'https://maps.nccs.nasa.gov/arcgis/home/item.html?id=ff4478ca84d24766bd79ac186bb60d9c#data'
-#    resp_glc = requests.get(url=url)
-
-#    url = 'https://data.nasa.gov/api/geospatial/h9d8-neg4?method=export&format=Shapefile'
-#    # Timeout error, website is currently not working
-#    LOGGER.info('requesting data from %s', url)
-#    response = requests.get(url=url)
-#    LOGGER.info('downloading content...')
-#    open((save_path+'/global_LS_catalogue'+'.zip'), 'wb').write(response.content)
-#
-
-def get_nowcast_tiff(tif_type="monthly", starttime="", endtime="", save_path=None):
-    """API request to get global monthly LS hazard map, averaged over 15 years from NASA
-    or daily global LS nowcasting hazard map. Both from NASA.
-
-    Paramters:
-        tif_type (str): monthly or daily;
-        starttime (str): 'yyyy-mm-dd' max. 90 days from now for daily LS tiff
-        endtime(str): 'yyyy-mm-dd' for daily LS tiff
-        save_path (str): save path for daily LS tiff (monthly gets downloaded into cwd)
-    Returns:
-        tiff files (monthly) to current_working_directory/
-            [01-12]_ClimatologyMonthly_032818_9600x5400.tif or
-        tiff files (daily) to save_path/LS_nowcast_date.tif
+def mapping_point2grid(geometry, ymax, xmin, res):
```

Comment:
Nope, this was exactly my question (see above in my comment to this pull request).

> There are 3 helper functions now in the landslide module. 2 for looking up the position of a point on a grid cell matrix, 1 for the binomial sampling from a probability matrix. - please advice whether they should be transferred to util

Would you advise to put it in the coordinates module?

Reply:
Ah yes, I remember, and this confused me. I think this would be a good idea indeed. As a matter of fact, just this morning I came across an application where I needed a function like `mapping_point2grid`.

Reply:
moved it.